src/restoreNames.test.js

```diff
@@ -1,11 +1,88 @@
 'use strict';
 
 describe('restoreNames', () => {
-  // const { restoreNames } = require('./restoreNames');
+  const { restoreNames } = require('./restoreNames');
 
-  it('should ', () => {
+  // write tests here
+  it(`should be declared`, () => {
+    expect(restoreNames).toBeInstanceOf(Function);
+  });
 
+  it(`should return nothing`, () => {
+    expect(restoreNames([])).toBeUndefined();
   });
 
-  // write tests here
+  it(`should throw Error if fullName not exist`, () => {
+    const users = [
+      {
+        firstName: undefined,
+        lastName: 'Holy',
+      },
+      {
+        lastName: 'Adams',
+      },
+    ];
+
+    expect(() => restoreNames(users)).toThrow();
+  });
+
+  it(`should restore firstName from fullName if it is undefined
```

Comment:
If the property is missing its also undefined so this description can be simplified.

```suggestion
  it(`should restore firstName from fullName if it is undefined`, () => {
```